sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -119,21 +117,24 @@ case class Size(child: Expression, legacySizeOfNull: Boolean)
     val value = child.eval(input)
     if (value == null) {
       if (legacySizeOfNull) -1 else null
-    } else child.dataType match {
-      case _: ArrayType => value.asInstanceOf[ArrayData].numElements()
-      case _: MapType => value.asInstanceOf[MapData].numElements()
-      case other => throw QueryExecutionErrors.unsupportedOperandTypeForSizeFunctionError(other)
-    }
+    } else
```

Comment:
Can we remove unrelated style changes?

Reply:
Yeah there were too many of those. I've removed them now.